CONTRIBUTING.adoc

```diff
@@ -5,225 +5,147 @@ OpenShift Developers <dev@lists.openshift.redhat.com>
-
-    # shut down openshift
-    $ sudo pkill openshift
-    # stop the docker containers
-    $ docker ps | awk 'index($NF,"k8s_")==1 { print $1 }' | xargs -l -r docker stop
-    # deleting all the internal config files, etcd, etc and starting openshift fresh
-    $ sudo rm -rf openshift.local.*
-    # if you used the --volume-dir=/home/vagrant/volumes flag, then run these
-
-TIP: See https://github.com/openshift/vagrant-openshift for more advanced options
-
-==== Ensure virtual box interfaces are not managed by Network Manager
-
-If you are developing on a Linux host, then you need to ensure that Network Manager is ignoring the
-virtual box interfaces, otherwise they cause issues with multi-vm networking.
-
-Follow these steps to ensure that virtual box interfaces are unmanaged:
-
-1. Check the status of Network Manager devices:
-
-   $ nmcli d
-
-2. If any devices whose name start with vboxnet* are not unmanaged, then they need to be added to
-   NetworkManager configuration to be ignored.
-
-   $ cat /etc/NetworkManager/NetworkManager.conf
-
-        [keyfile]
-        unmanaged-devices=mac:0a:00:27:00:00:00;mac:0a:00:27:00:00:01;mac:0a:00:27:00:00:02
-
-3. One can use the following command to help generate the configuration:
-
-   $ ip link list | grep vboxnet  -A 1 | grep link/ether | awk '{print "mac:" $2}' |  paste -sd ";" -
-
-4. Reload the Network Manager configuration:
-
-    $ sudo nmcli con reload
+3. Open up a new terminal or source the changes in your current terminal.  Then
+   clone your forked repo:+
+----
+$ mkdir -p $GOPATH/src/github.com/openshift
+$ cd $GOPATH/src/github.com/openshift
+$ git clone git://github.com/<forkid>/origin  # Replace <forkid> with the your github id
+$ cd origin
+$ git remote add upstream git://github.com/openshift/origin
+----
+4.  From here, you can generate the OpenShift binaries and images by running:+
+----
+$ make release
+----
+    If you want to generate just the OpenShift binaries, you could run:+
+----
+$ make build
+----
+    Or, if you did not install `go` you can build the server binary in our
+    release environment:+
+----
+$ hack/env make build
+----
+    Then, you can use the same environment to build client binaries for your
+    local system. For example, on Mac OS:+
+----
+$ hack/env OS_BUILD_PLATFORMS=darwin/amd64 make build WHAT=cmd/oc
+----
+5.  If it is not there already, add the current build output directory to the
+    `$PATH`, so you can leverage the OpenShift commands elsewhere:+
+----
+$ export PATH="${PATH}:$( source hack/lib/init.sh; echo "${OS_OUTPUT_BINPATH}$( os::util::host_platform )/" )"
+----
+6.  Next, follow the set-up steps in ./docs/cluster_up_down.md[`cluster_up_down.md`]
+    to start a cluster with `oc cluster up`. When starting the cluster, you
+    will need to use the container images you built in step 4, which are
+    tagged with the `git` commit you're working off of:+
+----
+$ oc cluster up --version="$(git log -1 --pretty=%h )"
```

Comment:
hm.  you'd think so, but that's not what i've ever observed when building new local sti-builder images and then running with latest images.

@csrwng theories/explanation?  (i know it's irrelevant for origin:latest since that's not a pod, but for the other infrastructure images, i'd think "oc cluster up --version=latest" would result in always-pull behavior for the deployer and builder infrastructure images)